client/src/components/Form/index.js

```diff
@@ -0,0 +1,122 @@
+import React, { Component } from "react";
+import axios from "axios"; 
+import "./Form.css";
+
+class Form extends Component {
+  // Setting the initial values of this.state.username and this.state.password
+  state = {
+    first_name: "",
+    last_name: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+    uuid: ""
+  };
+
+  // handle any changes to the input fields
+  handleInputChange = event => {
+    // Pull the name and value properties off of the event.target (the element which triggered the event)
+    const { name, value } = event.target;
+
+    // Set the state for the appropriate input field
+    this.setState({
+      [name]: value
+    });
+  };
+
+  // When the form is submitted, prevent the default event and alert the username and password
+  handleRegisterSubmit = event => {
+    event.preventDefault();
+    // check if password matches
+    if (this.state.password !== this.state.confirmPassword) {
+      alert("Passwords don't match, please try again");
+      this.setState({ password: "", confirmPassword: "" });
+      return;
+    }
+    // send user to 
+    axios.post("/auth/register", {
+      first_name: this.state.first_name,
+      last_name: this.state.last_name,
+      email: this.state.email,
+      password: this.state.password,
+      confirmPassword: this.state.confirmPassword
+    }).then(res => {
+      if(res.data.saved){
+        
+        this.setState({ 
+          first_name: "",
+          last_name: "",
+          email: "",
+          password: "",
+          confirmPassword: "",
+          uuid: res.data.uuid
+        });
+        alert("Thank you for registering!")
+      } else if (!res.data.saved) {
+        this.setState({
+          first_name: "",
+          last_name: "",
+          email: "",
+          password: "",
+          confirmPassword: ""
+        });
```

Comment:
It there any reason for setting state here other than clearing the form? 

In the event the user is successful we could place this form inside of a parent div and if saved, do a `document.getElementById("divname").innerHTML = "Thank you for registering, to activate check your email"` or we should generate an access token for them and pass this into the database before send it up to be stored in localStorage before redirecting them back to the page that they had come from. Once redirected back, it should trigger the decoding of that token, check the db and "auto" log them in post registration.

Definitely understand that this is just a way to test saving users, but might be a good callout for future features.

Reply:
Yes, currently it is being set to clear the form.  I don't think the form is set up quite right. I believe I should be passing everything to the form via props and the whole state should live on the register page.  This way we can use the form in other places if desired. 

Right now I only put in an alert that they have registered but I was thinking we should just automatically redirect them back to the page they were one.  I like your idea which I believe is just clearing out the whole form and putting "Thank you for registering, to activate check your email" or were you thinking just adding it to the top of the page?  

I agree we should also implement some sort of automatic login.  Should we do this as part of the registration? or wait until the frontend gets the response that it was saved then send a request to the login route?  We should also decide how we want to save the uuid. or do we even need it if they automatically get logged in?

Reply:
Sounds good! I think you're right here with the passing of props into the component so it gives us wider usability. In this particular case I don't think that it is too worrisome as register is it's own page so it likely isn't being pulled in as a component in many cases outside of this. If we were to instead create separate form components for things like `input` or `button` then 100% props would be the route to go. 

Outside of this, I think if we go the redirect route after registration we should be good to go! On register we would just need to create a JWT token update the database to store that token and then pass it upward so it can be stored. To note, I was doing some reading last night though and I think I might want to drop the use of localStorage as anyone could come in, get the token that is stored and set it in their browser in order to act as the user that logged in (so long as that token has not expired). Might need to explore the use of cookies, or drop auto login altogether as it seems rather insecure the way that it is set up now.